be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/WitnessValidator.scala

```diff
@@ -24,20 +24,15 @@ sealed class WitnessValidator(dbActorRef: => AskableActorRef) extends MessageDat
     rpcMessage.getParamsMessage match {
       case Some(message: Message) =>
         val data: WitnessMessage = message.decodedData.get.asInstanceOf[WitnessMessage]
```

Comment:
You probably want to use extractData[T](), this method return (t: T, hash: Hash, senderPK: PublicKey, channel: Channel), it will make the code more concise.